knowledge-base/microsoft/psscriptanalyzer-action/action-security.yml

```diff
@@ -0,0 +1,14 @@
+name: Run PSScriptAnalyzer
+outbound-endpoints:
+  # NOTE: https://docs.microsoft.com/en-us/powershell/scripting/gallery/getting-started?view=powershell-7.2#network-access-to-the-powershell-gallery
+  - fqdn: www.powershellgallery.com
+    port: 443
+    reason: to fetch powershell modules
+  
+  - fqdn:  psg-prod-eastus.azureedge.net.
+    port: 443
+    reason: to download powershell modules
+
+  - fqdn: az818661.vo.msecnd.net
```

Comment:
@h0x0er did you run `harden-runner` on this? Don't see `az818661.vo.msecnd.net` or other domains in the action source code.

Reply:
@varunsh-coder Yes i ran the harden-runner,  but somehow the `insights` are not visible [see_this](https://app.stepsecurity.io/github/h0x0er/kb_setup/actions/runs/1706424205). 
As per [this](https://docs.microsoft.com/en-us/powershell/scripting/gallery/getting-started?view=powershell-7.2#network-access-to-the-powershell-gallery), powershell gallery uses some cdns to download the modules, so added those cdn domains.

Reply:
The repo needs to be public. Harden-runner doesn’t work for private repos as of now.

Reply:
I will fix that thing and make a new commit with added `harden-runner-link`